aws/resource_aws_sagemaker_notebook_instance.go

```diff
@@ -0,0 +1,371 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+	"time"
+)
+
+func resourceAwsSagemakerNotebookInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSagemakerNotebookInstanceCreate,
+		Read:   resourceAwsSagemakerNotebookInstanceRead,
+		Update: resourceAwsSagemakerNotebookInstanceUpdate,
+		Delete: resourceAwsSagemakerNotebookInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Computed:     false,
+				ForceNew:     true,
+				ValidateFunc: validateSagemakerName,
+			},
+
+			"role_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				Computed: false,
+			},
+
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"security_groups": {
+				Type:     schema.TypeSet,
+				MinItems: 1,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"creation_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"last_modified_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsSagemakerNotebookInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	var name string
+	if v, ok := d.GetOk("name"); !ok {
+		return fmt.Errorf("Error reading Sagemaker Notbook Instance name")
+	} else {
+		name = v.(string)
+	}
+
```

Comment:
This else clause is not needed since we return in the error case anyway.

Reply:
Fixed, I'll fix the training job as well